pkg/types/openstack/machinepool.go

```diff
@@ -55,12 +55,9 @@ func (o *MachinePool) Set(required *MachinePool) {
 		o.AdditionalSecurityGroupIDs = append(required.AdditionalSecurityGroupIDs[:0:0], required.AdditionalSecurityGroupIDs...)
 	}
 
-	if len(required.Zones) > 0 {
+	if required.Zones != nil {
```

Comment:
zones is a string slice, so you should be checking for len instead of nil

Reply:
+1 if a user doesnt enter a zone, it will return empty slice. We should adjust the validations and downstream assets to be able to handle receiving an empty slice, and to translate that into zone == "" (which is nova default)

Reply:
done!